Feat/cp 25 third party sign in

Benchen997 opened the pull request:
1. use passport strategy to implement oAuth login for google and facebook
2. google login is avaliable for use now
3. facebook service is ready, but wait for future client key and secrete
4. user with oAuth login who have email registrated as our previous user will be treated as same user

Chris-Wang reviewed (changes requested):
Please squash your commits into one

Commit:
feat/CP-25/Third-Party-Login
Added Oauth service for google and facebook:
- implemented google and facebook strategy
- implemented auth controller to receive redirect, accept access token from Oauth provider
- implemented handleThirdParty login function in auth.service.ts

update: env variables, including google client secret and client Id

merge conflict:
- git pull origin dev
- git checkout CP-25
- git merge dev
- resolve conflict from email service, auth module.
- git push

Resolve CP-25

Chris-Wang commented:
let me recheck this when our google account get fixed

Commit:
Merge branch 'dev' into feat/CP-25-third-party-sign-in

Commit:
fix prettier formatting

Commit:
remove console logger for user token

Commit:
Merge branch 'dev' into feat/CP-25-third-party-sign-in
